### PR TITLE
Close #12164: Use Coords for place_virtual_track

### LINKED ISSUE #12164: Use Coords for TrackDesign functions
Some TrackDesign functions still take individual `x` and `y` parameters. The goal is to make them use `CoordsXY*`.

Note: You can do all of the functions in all PR (seems small overall), but please do each of them into a different commit.

- [x] - `track_design_update_max_min_coordinates` should receive a `const CoordsXYZ&` instead of `x`, `y` and `z`.
- [x] - `track_design_add_selection_tile` should receive a `const CoordsXY&` instead of `x` and `y`.
- [x] - `track_design_place_maze` should receive a `const CoordsXYZ&` instead of `x`, `y` and `z`.
- [ ] - `track_design_place_ride` should receive a `const CoordsXYZ&` instead of `x`, `y` and `z`.
- [ ] - `place_virtual_track` should receive a `const CoordsXYZ&` instead of `x`, `y` and `z`.

Functions are at `src\openrct2\ride\TrackDesign.cpp`

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

File: src/openrct2/ride/TrackDesign.cpp [hunk 1]
```diff
@@ -1935,7 +1930,7 @@ static bool track_design_place_preview(TrackDesign* td6, money32* cost, Ride** o
     int32_t mapSize = gMapSize << 4;
 
     _currentTrackPieceDirection = 0;
-    int32_t z = place_virtual_track(td6, PTD_OPERATION_GET_PLACE_Z, true, GetOrAllocateRide(0), mapSize, mapSize, 16);
+    int32_t z = place_virtual_track(td6, PTD_OPERATION_GET_PLACE_Z, true, GetOrAllocateRide(0), { mapSize, mapSize, 16 });
```
Comment (Gymnasiast) [inline src/openrct2/ride/TrackDesign.cpp:1933]:
```suggestion
    int32_t z = place_virtual_track(td6, PTD_OPERATION_GET_PLACE_Z, true, GetOrAllocateRide(0), { mapSize, mapSize, MINIMUM_LAND_HEIGHT_BIG });
```

File: src/openrct2/ride/TrackDesign.cpp [hunk 2]
```diff
@@ -333,7 +333,7 @@ rct_string_id TrackDesign::CreateTrackDesignTrack(const Ride& ride)
         }
     }
 
-    place_virtual_track(this, PTD_OPERATION_DRAW_OUTLINES, true, GetOrAllocateRide(0), 4096, 4096, 0);
+    place_virtual_track(this, PTD_OPERATION_DRAW_OUTLINES, true, GetOrAllocateRide(0), { 4096, 4096, 0 });
```
Comment (Gymnasiast) [inline src/openrct2/ride/TrackDesign.cpp:336]:
```suggestion
    place_virtual_track(this, PTD_OPERATION_DRAW_OUTLINES, true, GetOrAllocateRide(0), { 128 * COORDS_XY_STEP, 128 * COORDS_XY_STEP, 0 });
```

Also apply this to line 455.

**Comment by Gymnasiast** [comment]
@tupaschoal Why did you merge this while my line notes were still open?

**Comment by tupaschoal** [comment]
I merged at the same time you commented @Gymnasiast , sorry.

**Comment by Gymnasiast** [comment]
@frutiemax Could you open a new PR to address the line notes?

**Pull request merged**